setup.py

```diff
@@ -22,9 +22,10 @@
 
 def load_msvcvar():
     if os.environ.get('vcvars'):
+        os.environ['__vcvars'] = '"{}"'.format(os.environ['vcvars'])
         stdout, _ = subprocess.Popen([
-            'cmd', '/q', '/c', '(%vcvars% & set)'],
-            stdout=subprocess.PIPE, shell=True, universal_newlines=True).communicate()
+            'cmd', '/q', '/c', 'call %__vcvars% && set'],
+            stdout=subprocess.PIPE, shell=False, universal_newlines=True).communicate()
```

Comment:
Does it need to create a new environment variable, or would just setting `shell=False` be enough?

Reply:
I think shell=True is to create another cmd to execute the command line 'cmd /q /c ...', which looks redundant.